QSPIFBlockDevice.cpp

```diff
@@ -0,0 +1,1226 @@
+        case 0xbf:
+            // SST devices come preset with block protection
+            // enabled for some regions, issue write disable instruction to clear
+            _set_write_enable();
+            _qspi_send_general_command(QSPIF_WRDI, -1, NULL, 0, NULL, 0);
+            break;
+    }
+
+    //Synchronize Device
+    if ( false == _is_mem_ready()) {
+        tr_error("ERROR: init - _is_mem_ready Failed");
+        status = QSPIF_BD_ERROR_READY_FAILED;
+        goto exit_point;
+    }
+
+    /**************************** Parse SFDP Header ***********************************/
+    if ( 0 != _sfdp_parse_sfdp_headers(basic_table_addr, basic_table_size, sector_map_table_addr, sector_map_table_size)) {
+        tr_error("ERROR: init - Parse SFDP Headers Failed");
+        status = QSPIF_BD_ERROR_PARSING_FAILED;
+        goto exit_point;
+    }
+
+
+    /**************************** Parse Basic Parameters Table ***********************************/
+    if ( 0 != _sfdp_parse_basic_param_table(basic_table_addr, basic_table_size) ) {
+        tr_error("ERROR: init - Parse Basic Param Table Failed");
+        status = QSPIF_BD_ERROR_PARSING_FAILED;
+        goto exit_point;
+    }
+
+    /**************************** Parse Sector Map Table ***********************************/
+    _region_size_bytes[0] =
+        _device_size_bytes; // If there's no region map, we have a single region sized the entire device size
+    _region_high_boundary[0] = _device_size_bytes - 1;
+
+    if ( (sector_map_table_addr != NULL) && (0 != sector_map_table_size) ) {
+        tr_info("INFO: init - Parsing Sector Map Table - addr: 0x%xh, Size: %d", sector_map_table_addr,
+                sector_map_table_size);
+        if (0 != _sfdp_parse_sector_map_table(sector_map_table_addr, sector_map_table_size) ) {
+            tr_error("ERROR: init - Parse Sector Map Table Failed");
+            status = QSPIF_BD_ERROR_PARSING_FAILED;
+            goto exit_point;
+        }
+    }
+
+    // Configure  BUS Mode to 1_1_1 for all commands other than Read
+    _qspi_configure_format(QSPI_CFG_BUS_SINGLE, QSPI_CFG_BUS_SINGLE, QSPI_CFG_ADDR_SIZE_24, QSPI_CFG_BUS_SINGLE,
+                           QSPI_CFG_ALT_SIZE_8, QSPI_CFG_BUS_SINGLE, 0);
+
+    _is_initialized = true;
+
+exit_point:
+    _mutex.unlock();
+
+    return status;
+}
+
+int QSPIFBlockDevice::deinit()
+{
+    _mutex.lock();
+    if (_is_initialized == false) {
+        _mutex.unlock();
+        return QSPIF_BD_ERROR_OK;
+    }
+
+    // Disable Device for Writing
+    qspi_status_t status = _qspi_send_general_command(QSPIF_WRDI, -1, NULL, 0, NULL, 0);
+    int result = QSPIF_BD_ERROR_OK;
+
+    if (status != QSPI_STATUS_OK)  {
+        tr_error("ERROR: Write Disable failed");
+        result = QSPIF_BD_ERROR_DEVICE_ERROR;
+    }
+    _is_initialized = false;
+    _mutex.unlock();
+
+    return result;
+}
+
+
+int QSPIFBlockDevice::read(void *buffer, bd_addr_t addr, bd_size_t size)
+{
+
```

Comment:
can you remove empty lines like this, 247-248 at least

Reply:
done for entire file